eng/common/pipelines/templates/jobs/sdk-generation.yml

```diff
@@ -0,0 +1,160 @@
+parameters:
+  - name: SpecRepoBranch
+    type: string
+    default: 'main'
+  - name: SpecRepoCommit
+    type: string
+    default: 'HEAD'
+  - name: SdkRepoOwner
+    type: string
+  - name: SdkRepoName
+    type: string
+
+  - name: SdkRepoNameForSpecGen
+    type: string
+  - name: SdkRepoBranch
+    type: string
+    default: 'main'
+  - name: ConfigType
```

Comment:
Could you remove this parameter and determine the type by the value of 'TspConfigPath' and 'ReadmePath'. And prioritize 'TspConfigPath' over 'ReadmePath'.

Reply:
DevOps does not really have a way of making parameters optional. so, I added this extra parameter to help determine which config the user intends to use. Otherwise, we will have to just have one configPath then determine which in the logic. I think this is clearer though that way we can still show the defaults for each config type.

Reply:
The first value becomes the default in this case "TypeSpec" so I think this is fine.